tasks/tangd-custom-port.yml

```diff
@@ -0,0 +1,65 @@
+---
+# This task tells SELinux for the port that we want the tangd service to use when distribution Fedora
+- name: Allow the custom port for tangd_port_t in SELinux
+  ansible.builtin.import_role:
+    name: linux-system-roles.selinux
+  vars:
+    selinux_ports:
+      - { ports: '{{ nbde_server_port }}', proto: 'tcp', setype: 'tangd_port_t', state: 'present' }
+  when: ansible_selinux.status == "enabled" and ansible_facts["distribution"] == "Fedora"
+
+# This task tells SELinux for the port that we want the tangd service to use when distribution RHEL
```

Comment:
You don't need this.  In fact, you don't need the `when: ansible_selinux.status == "enabled" and ansible_facts["distribution"] == "Fedora"` clause above.

The problem is that `ansible_facts` will return the distribution on the **managed host**, but the roles will be looked up on the **controller host**.  I think the logic you are hoping for is if the **controller host** is `RedHat`, then use the rhel-system-roles.selinux role, but that will be much trickier to do.

And, to further complicate matters, if the user is using the Collection instead of the legacy role format, then the role name will be `fedora.linux_system_roles.selinux` or `redhat.rhel_system_roles.selinux`.

I think the best you can do here is to use `import_role: name: selinux` and hope that role path and collection path/collection keyword are configured appropriately.

Reply:
Just use `name: linux-system-roles.selinux` - we have tooling in the rhel rpm spec to change it to `rhel-system-roles` and similar tooling when we build collections.